src/OrchardCore.Modules/OrchardCore.Html/Drivers/HtmlBodyPartDisplay.cs

```diff
@@ -56,7 +62,7 @@ public override async Task<IDisplayResult> UpdateAsync(HtmlBodyPart model, IUpda
                 }
                 else
                 {
-                    model.Html = viewModel.Html;
+                    model.Html = (settings.AllowCustomScripts) ? viewModel.Html : _htmlScriptSanitizer.Sanitize(viewModel.Html);
```

Comment:
This shold be done in BuildViewModel otherwise you can still inject it with liquid.

Reply:
Which means it's at render time. Which is an issue as it will impact perf. The other solution would be to not accept liquid, and instead use custom bbcode tags that are safe, and then we could do it at that point.

Reply:
Even at render time, liquid can still inject scriptblocks, or zones, which would not be in the output of the liquid here.

I tried over the weekend and was able to work around any sanitization with liquid.

So maybe bbcode only ?

Reply:
It is done in the `UpdateAsync`.

Reply:
> Even at render time, liquid can still inject scriptblocks

What I suggested is to sanitized after Liquid is evaluated, in the BuildViewModel. Though anyone using custom templates without the shape result would also need to do it, with a custom filter. Which should be implemented anyhow. I'd like to move to bbcodes, yes. Intead of Liquid for these kinds of parts. The idea would be use Html/Markdown fields and parts with bbcode (sanitization would be done at update time), and if you need advanced logic then use the Liquid part. We could decide to implement the sanitization in Liquid parts too, and it would be at render time.